Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-147.ca-central-1.compute.internal
Sha: 70513782806a88de56747fc3bf69875656ba017b
Build: success
Build: failed!
Number of Error(s): 12
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:49 (file):
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:56 (file):
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:63 (file):
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:70 (file):
CPack Error: Error when generating package: hpccsystems-platform
make: *** [Makefile:130: package] Error 1
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:49 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core/libaws-cpp-sdk-core.so
  No valid ELF RPATH or RUNPATH entry exists in the file;
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14093/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14093/build/cmake_install.cmake:46 (include)
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:56 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-sqs/libaws-cpp-sdk-sqs.so
  The current RPATH is:
    /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:
  which does not contain:
    /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:/home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/.deps/install/lib64:
  as was expected.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14093/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14093/build/cmake_install.cmake:46 (include)
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:63 (file):
  file RPATH_CHANGE could not write new RPATH:
    /opt/HPCCSystems/lib
  to the file:
    /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-s3/libaws-cpp-sdk-s3.so
  The current RPATH is:
    /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:
  which does not contain:
    /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/aws-cpp-sdk-core:/home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/.deps/install/lib64:
  as was expected.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14093/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14093/build/cmake_install.cmake:46 (include)
CMake Error at /home/centos/smoketest/PR-14093/build/system/aws/cmake_install.cmake:70 (file):
  file INSTALL cannot find
  /home/centos/smoketest/PR-14093/build/system/aws/aws-sdk-cpp/.deps/install/lib64/libaws-c-common.so.1.0.0.
Call Stack (most recent call first):
  /home/centos/smoketest/PR-14093/build/system/cmake_install.cmake:46 (include)
  /home/centos/smoketest/PR-14093/build/cmake_install.cmake:46 (include)
CPack Error: Error when generating package: hpccsystems-platform
make: *** [Makefile:130: package] Error 1
Package end (2020-08-19_10-50-50 6 se
 ... ( comment is too long, 99 lines, 5693 bytes, truncated) ... 
[Build error:](https://gist.github.com/60ab6affa28766838dbd59b8b460e508)
Build log [Build log](https://gist.github.com/HPCCSmoketest/60ab6affa28766838dbd59b8b460e508#file-result-20-08-19-10-41-04-log)

```

--- a/.github/workflows/test-helm.yml
+++ b/.github/workflows/test-helm.yml
@@ -40,6 +40,9 @@ jobs:
         with:
           version: 'latest'
         id: install
+      - name: Install additional tools
+        working-directory: .
+        run: wget https://github.com/zegl/kube-score/releases/download/v1.8.1/kube-score_1.8.1_linux_amd64 && sudo mv kube-score_1.8.1_linux_amd64 /usr/bin/kube-score && sudo chmod +x /usr/bin/kube-score
       - name: Test
         working-directory: ./testing/helm
         run: ./run.sh

--- a/testing/helm/run.sh
+++ b/testing/helm/run.sh
@@ -41,4 +41,30 @@ do
       cat results.txt
    fi
 done
+
+if type kubeval >/dev/null 2> /dev/null; then
+   echo Running kubeval...
+   helm template $hpccchart ${options} | kubeval --strict - >results.txt 2>errors.txt
+   if [ $? -ne 0 ]
+   then
+      echo $file failed
+      cat errors.txt
+      cat results.txt
+      failed=1
+   fi
+fi
+
+if type kube-score >/dev/null 2> /dev/null; then
+   echo Running kube-score...
+   # Note we force all replicas to be > 1 as some checks are not done on replicas=1 cases e.g. antiaffinity
+   helm template $hpccchart ${options} | sed "s/replicas: 1/replicas: 2/" | kube-score score --output-format ci - >results.txt 2>errors.txt
+   if [ $? -ne 0 ]
+   then
+      echo $file failed
+      cat errors.txt
+      cat results.txt
+      failed=1
+   fi
+fi
+
 exit $failed

